### PR TITLE
content: add list how-to guide

### LINKED ISSUE #112: How to: client.list
Add a how to section for the new client.list method. 

The new `client.list` method was added in [this PR](https://github.com/web3-storage/web3.storage/pull/198). It uses the `GET /user/uploads` http endpoint under the hood. Oli updated the browser demo code to use it too, [here](https://github.com/web3-storage/web3.storage/pull/222). 

Acceptance criteria: 
- [x]  There's a how to page for client.list
- [x]  Code samples can be copied and pasted
- [x]  Covers additional topics the user might need to think about when using this particular api method. Stuff like *"when storing data, you might want to consider encrypting it using GPG",* or, *"consider splitting your data into chunks before storing to make retrieval easier",* etc.
- [x]  Format follows our guidelines
- [x]  Has been copy edited by someone who didn't write it
- [ ]  Has been tested by someone who didn't write it
- [ ]  Generate API section is complete
- [ ]  Has been tech proofed by Oli, Alan, or Vasco

**THIS DOES NOT HAVE TO HIT LAUNCH DEADLINE.**

### PR DESCRIPTION
**Pull request by yusefnapora**
This add a how-to guide for listing your uploads using the JS client, with a mention that you can just view things on the files page as well.

closes #112

### CODE REVIEW

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/list.md`:
- The following link returned HTTP status code 302 (Found):
  https://web3.storage/files
- Did you mean “them”?
  ```
  ...opens new window) that can be used with the for await syntax  (opens new win...
                                             ^^^
  ```
- Possible agreement error.
  ```
  ...an the total size of the files on disk. pins contains an array of objects de...
                                             ^^^^^^^^^^^^^
  ```
- Possible agreement error.
  ```
  ... fast retrieval using the IPFS network. deals contains an array of objects d...
                                             ^^^^^^^^^^^^^^
  ```
- Two determiners in a row. Choose either “the” or “the”.
  ```
  ... you can query an upload's status using the the cid.  By default the list me...
                                             ^^^^^^^
  ```
- Did you mean: “By default,”?
  ```
  ... an upload's status using the the cid.  By default the list method returns i...
                                             ^^^^^^^^^^
  ```

File: docs/how-tos/list.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+---
+title: List
+description: Learn how to list the files you've uploaded to Web3.Storage
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:3]:
```suggestion
description: Learn how to list the files you've uploaded to Web3.Storage in this quick how-to guide.
```
(for consistency with other how-tos)

File: docs/how-tos/list.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+---
+
+# How to list files uploaded to Web3.Storage
+
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:7]:
```suggestion
In this how-to guide, you'll learn about the different ways that you can **list the files that you've uploaded to Web3.Storage.**
```

(for consistency with other how-tos)

File: docs/how-tos/list.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+Once you've [stored some files][howto-store] using Web3.Storage, you can see a list of everything you've uploaded on the [Files page on the Web3.Storage website][site-files].
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:8]:
```suggestion
Once you've [stored some files][howto-store] using Web3.Storage, you'll want to see a list of what you've uplodaded. There are two ways you can do this:
- Using the 

## Using the Web3.Storage website

You can see a list of everything you've uploaded to Web3.Storage on the [Files page][site-files] on the Web3.Storage website. If you don't need to work with this list programmatically, using the website may be a simpler choice.
```
Adding some introductory comment to emphasize:
- there are two ways you can list files
- the programmatic one is preferable unless you're just looking for something simple

File: docs/how-tos/list.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+![A screenshot of the file listing available at https://web3.storage/files when logged in to your account](../images/files-listing.png)
+
+The [Files page][site-files] provides a convenient overview of your stored data, with links to view your files using an [IPFS gateway][ipfs-docs-gateway] and information about how the data is being stored on the [decentralized storage networks][concepts-decentralized-storage] that Web3.Storage uses under the hood.
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:12]:
```suggestion
This [Files page][site-files] provides a convenient overview of your stored data, including links to view your files in your browser via an [IPFS gateway][ipfs-docs-gateway] and information about how the data is being stored on the [decentralized storage networks][concepts-decentralized-storage] that Web3.Storage uses under the hood.
```

File: docs/how-tos/list.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:13]:
```suggestion
## Using the Web3.Storage client
```

File: docs/how-tos/list.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+You can also access a listing of your uploads from your application code using the Web3.Storage client. This guide will show you how to use the [JavaScript client library][reference-js-client] to fetch a complete listing of all the data you've uploaded using Web3.Storage.
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:14]:
```suggestion
To easily integrate Web3.Storage programmatically in your apps or services, you can also access a listing of your uploads from your code using the Web3.Storage client. In the example below, this guide walks through how to use the [JavaScript client library][reference-js-client] to fetch a complete listing of all the data you've uploaded using Web3.Storage.
```

File: docs/how-tos/list.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+## Installing the client
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:16]:
```suggestion
### Installing the client
```

File: docs/how-tos/list.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+In your JavaScript project, add the `web3.storage` package to your dependencies:
+
+```shell
+npm install web3.storage
+```
+
+## Creating a client instance
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:24]:
```suggestion
### Creating a client instance
```

File: docs/how-tos/list.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+To create a `Web3Storage` client object, we need to pass an access token into the [constructor][reference-js-constructor]:
+
+<<<@/code-snippets/how-to/index.js#makeStorageClient
+
+:::tip
+You can use any access token associated with your account, not just the one you used to upload your files! See the [Generate API token page][howto-gen-token] for more about token management.
+:::
+
+## Listing your uploads
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:34]:
```suggestion
### Listing your uploads
```

File: docs/how-tos/list.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+- `dagSize` contains the size in bytes of the [Directed Acyclic Graph (DAG)][ipfs-docs-merkle-dag] that contains all the uploaded content. This is the size of the data that is transferred over the network to Web3.Storage during upload, and is slightly larger than the total size of the files on disk.
+- `pins` contains an array of objects describing the IPFS nodes that have [pinned][ipfs-docs-pinning] the data, making it available for fast retrieval using the IPFS network.
+- `deals` contains an array of objects describing the Filecoin storage providers that have made [storage deals][fil-docs-deals]. These storage providers have committed to storing the data for an agreed period of time.
+
+::: tip "Want more details about storage?"
+The `Upload` objects returned by the `list` method include some basic status information about how the data is stored on IPFS and Filecoin. For more details, including the identity of the storage providers hosting your data, you can [query an upload's status][howto-query] using the the `cid`.
+:::
+
+### Listing a subset of uploads
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:61]:
```suggestion
#### Listing a subset of uploads
```

File: docs/how-tos/list.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+<<<@/code-snippets/how-to/index.js#listWithLimits
+
+[howto-store]: ./store.md
+[howto-retrieve]: ./retrive.md
+[howto-query]: ./query.md
+[howto-gen-token]: ./generate-api-token.md
+[concepts-decentralized-storage]: ../concepts/decentralized-storage.md
+[reference-js-client]: ../../reference/client-library.md
+
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:75]:
```suggestion
[reference-js-constructor]: ../reference/client-library.md#constructor
```
to fix a broken link above

File: docs/how-tos/list.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+```
+
+## Creating a client instance
+
+To create a `Web3Storage` client object, we need to pass an access token into the [constructor][reference-js-constructor]:
+
+<<<@/code-snippets/how-to/index.js#makeStorageClient
+
+:::tip
+You can use any access token associated with your account, not just the one you used to upload your files! See the [Generate API token page][howto-gen-token] for more about token management.
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:31]:
```suggestion
You can use any API token associated with your account, not just the one you originally used to upload your files! See the [Generate API token page][howto-gen-token] for more about token management.
```

File: docs/how-tos/list.md [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+:::
+
+## Listing your uploads
+
+The `Web3Storage` client object's `list` method returns an [async iterable][js-async-iterable-explainer] that can be used with the [`for await` syntax][mdn-for-await-of] to read information about each upload as soon as it's received over the network.
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:36]: Can you please add a link for ``'list' method` once it gets added to the client library page?

File: docs/how-tos/list.md [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+- `cid` contains the [IPFS Content Identifier (CID)][ipfs-docs-cid] that identifies the uploaded data. This CID can be used to [retrieve][howto-retrieve] the uploaded files or get more detailed [status information][howto-query].
+- `created` contains an [ISO-8601 datetime string][iso-8601] indicating when the content was first uploaded to Web3.Storage.
+- `dagSize` contains the size in bytes of the [Directed Acyclic Graph (DAG)][ipfs-docs-merkle-dag] that contains all the uploaded content. This is the size of the data that is transferred over the network to Web3.Storage during upload, and is slightly larger than the total size of the files on disk.
+- `pins` contains an array of objects describing the IPFS nodes that have [pinned][ipfs-docs-pinning] the data, making it available for fast retrieval using the IPFS network.
+- `deals` contains an array of objects describing the Filecoin storage providers that have made [storage deals][fil-docs-deals]. These storage providers have committed to storing the data for an agreed period of time.
+
+::: tip "Want more details about storage?"
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:57]:
```suggestion
::: tip Want more details about storage?
```
No quotes necessary 😊

File: docs/how-tos/list.md [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+The `Upload` objects returned by the `list` method include some basic status information about how the data is stored on IPFS and Filecoin. For more details, including the identity of the storage providers hosting your data, you can [query an upload's status][howto-query] using the the `cid`.
+:::
+
+### Listing a subset of uploads
+
+By default the `list` method returns information about all uploads made using your Web3.Storage account. You can optionally restrict the listing to contain entries that were uploaded before a given timestamp, and you can also limit the total number of returned entries.
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:63]: Can you please add a link for ``'list' method` once it gets added to the client library page?

Comment (jessicaschilling) [inline docs/how-tos/list.md:63]:
```suggestion
By default. the `list` method returns information about all uploads made using your Web3.Storage account. You can optionally restrict the listing in two ways:
-  Only contain entries that were uploaded before a given timestamp.
- Limit the total number of returned entries.
```

File: docs/how-tos/list.md [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+
+Here's an example of fetching the first ten uploads made on the previous day:
```
Comment (jessicaschilling) [inline docs/how-tos/list.md:65]:
```suggestion
Here's an example of fetching the first 10 uploads made on the previous day:
```

**Review by jessicaschilling — CHANGES_REQUESTED** [review]
Thank you! Made some suggestions for consistency with other guides, and more clarity on web vs code approaches.

One other thing: Can you please add links where appropriate (noted in comments) to the list method when it exists in the JS client library page? Thank you!

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/list.md`:
- The following link returned HTTP status code 302 (Found):
  https://web3.storage/files
- Possibly missing preposition found.
  ```
  ...nce you've stored some files using Web3.Storage, you'll want to see a list o...
                                             ^^^^^^^
  ```
- Possible typo. Did you mean “the” or “You”?
  ```
  ...ere are two ways you can do this: Using the  You can see a list of everythin...
                                             ^^^^^^^^
  ```
- Did you mean “these”?
  ```
  ...g the website may be a simpler choice.  This Files page  (opens new window) ...
                                             ^^^^
  ```
- Did you mean “them”?
  ```
  ...opens new window) that can be used with the for await syntax  (opens new win...
                                             ^^^
  ```
- Possible agreement error.
  ```
  ...an the total size of the files on disk. pins contains an array of objects de...
                                             ^^^^^^^^^^^^^
  ```
- Possible agreement error.
  ```
  ... fast retrieval using the IPFS network. deals contains an array of objects d...
                                             ^^^^^^^^^^^^^^
  ```
- Two determiners in a row. Choose either “the” or “the”.
  ```
  ... you can query an upload's status using the the cid.  By default. the list m...
                                             ^^^^^^^
  ```
- Did you mean: “By default,”?
  ```
  ... an upload's status using the the cid.  By default. the list method returns ...
                                             ^^^^^^^^^^
  ```

**Comment by terichadbourne** [comment]
Changing our requested Spark reviewer from Vasco to @alanshaw due to vacation schedules. 🙏

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/list.md`:
- The following link returned HTTP status code 302 (Found):
  https://web3.storage/files
- Possibly missing preposition found.
  ```
  ...nce you've stored some files using Web3.Storage, you'll want to see a list o...
                                             ^^^^^^^
  ```
- Possible typo. Did you mean “the” or “You”?
  ```
  ...ere are two ways you can do this: Using the  You can see a list of everythin...
                                             ^^^^^^^^
  ```
- Did you mean “these”?
  ```
  ...g the website may be a simpler choice.  This Files page  (opens new window) ...
                                             ^^^^
  ```
- Did you mean “them”?
  ```
  ...opens new window) that can be used with the for await syntax  (opens new win...
                                             ^^^
  ```
- Possible agreement error.
  ```
  ...an the total size of the files on disk. pins contains an array of objects de...
                                             ^^^^^^^^^^^^^
  ```
- Possible agreement error.
  ```
  ... fast retrieval using the IPFS network. deals contains an array of objects d...
                                             ^^^^^^^^^^^^^^
  ```
- Two determiners in a row. Choose either “the” or “the”.
  ```
  ... you can query an upload's status using the the cid.  By default, the list m...
                                             ^^^^^^^
  ```

**Comment by yusefnapora** [comment]
Last commit adds links to the reference section from #121, so it's probably best to merge that one first.

File: docs/how-tos/list.md [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+
+The `Web3Storage` client object's [`list` method][reference-js-list] returns an [async iterable][js-async-iterable-explainer] that can be used with the [`for await` syntax][mdn-for-await-of] to read information about each upload as soon as it's received over the network.
+
+Here's an example that logs details about each upload to the console:
+
+<<<@/code-snippets/how-to/index.js#listUploads
+
+Each `Upload` object will look something like this:
+
+<!-- TODO: replace with example that includes deal status -->
```
Comment (johndmulhausen) [inline docs/how-tos/list.md:49]: You can grab the example from the client reference, I got one there:

```json
{
  "name": "cat pics",
  "cid": "bafkreifzjut3te2nhyekklss27nh3k72ysco7y32koao5eei66wof36n5e",
  "created": "2021-07-14T19:27:14.934572Z",
  "dagSize": 101,
  "pins": [{
    "peerId": "12D3KooWR1Js",
    "peerName": "peerName",
    "region": "peerRegion",
    "status": "Pinned"
  }],
  "deals": [{
    "dealId": 12345,
    "miner": "f99",
    "status": "Active",
    "pieceCid": "bafkreifzjut3te2nhyekklss27nh3k72ysco7y32koao5eei66wof36n5e",
    "dataCid": "bafkreifzjut3te2nhyekklss27nh3k72ysco7y32koao5eei66wof36n5e",
    "dataModelSelector": "Links/0/Links",
    "activation": "2021-07-14T19:27:14.934572Z",
    "created": "2021-07-14T19:27:14.934572Z",
    "updated": "2021-07-14T19:27:14.934572Z"
  }]
}
```

Comment (yusefnapora) [reply docs/how-tos/list.md:49]: good call, thanks 👍

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/list.md`:
- The following link returned HTTP status code 302 (Found):
  https://web3.storage/files
- Possibly missing preposition found.
  ```
  ...nce you've stored some files using Web3.Storage, you'll want to see a list o...
                                             ^^^^^^^
  ```
- Possible typo. Did you mean “the” or “You”?
  ```
  ...ere are two ways you can do this: Using the  You can see a list of everythin...
                                             ^^^^^^^^
  ```
- Did you mean “these”?
  ```
  ...g the website may be a simpler choice.  This Files page  (opens new window) ...
                                             ^^^^
  ```
- Did you mean “them”?
  ```
  ...opens new window) that can be used with the for await syntax  (opens new win...
                                             ^^^
  ```
- Possible agreement error.
  ```
  ...an the total size of the files on disk. pins contains an array of objects de...
                                             ^^^^^^^^^^^^^
  ```
- Possible agreement error.
  ```
  ... fast retrieval using the IPFS network. deals contains an array of objects d...
                                             ^^^^^^^^^^^^^^
  ```
- Two determiners in a row. Choose either “the” or “the”.
  ```
  ... you can query an upload's status using the the cid.  By default, the list m...
                                             ^^^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/list.md`:
- The following link returned HTTP status code 302 (Found):
  https://web3.storage/files
- Possibly missing preposition found.
  ```
  ...nce you've stored some files using Web3.Storage, you'll want to see a list o...
                                             ^^^^^^^
  ```
- Possible typo. Did you mean “the” or “You”?
  ```
  ...ere are two ways you can do this: Using the  You can see a list of everythin...
                                             ^^^^^^^^
  ```
- Did you mean “these”?
  ```
  ...g the website may be a simpler choice.  This Files page  (opens new window) ...
                                             ^^^^
  ```
- Did you mean “them”?
  ```
  ...opens new window) that can be used with the for await syntax  (opens new win...
                                             ^^^
  ```
- Possible agreement error.
  ```
  ...an the total size of the files on disk. pins contains an array of objects de...
                                             ^^^^^^^^^^^^^
  ```
- Possible agreement error.
  ```
  ... fast retrieval using the IPFS network. deals contains an array of objects d...
                                             ^^^^^^^^^^^^^^
  ```
- Two determiners in a row. Choose either “the” or “the”.
  ```
  ... you can query an upload's status using the the cid.  By default, the list m...
                                             ^^^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/list.md`:
- The following link returned HTTP status code 302 (Found):
  https://web3.storage/files
- Possibly missing preposition found.
  ```
  ...nce you've stored some files using Web3.Storage, you'll want to see a list o...
                                             ^^^^^^^
  ```
- Possible typo. Did you mean “the” or “You”?
  ```
  ...ere are two ways you can do this: Using the  You can see a list of everythin...
                                             ^^^^^^^^
  ```
- Did you mean “these”?
  ```
  ...g the website may be a simpler choice.  This Files page  (opens new window) ...
                                             ^^^^
  ```
- Did you mean “them”?
  ```
  ...opens new window) that can be used with the for await syntax  (opens new win...
                                             ^^^
  ```
- Possible agreement error.
  ```
  ...an the total size of the files on disk. pins contains an array of objects de...
                                             ^^^^^^^^^^^^^
  ```
- Possible agreement error.
  ```
  ... fast retrieval using the IPFS network. deals contains an array of objects d...
                                             ^^^^^^^^^^^^^^
  ```

**Review by alanshaw — CHANGES_REQUESTED** [review]

File: code-snippets/how-to/example-listing-upload.json [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+  "deals": [{
+    "dealId": 12345,
+    "miner": "f99",
+    "status": "Active",
+    "pieceCid": "bafkreifzjut3te2nhyekklss27nh3k72ysco7y32koao5eei66wof36n5e",
+    "dataCid": "bafkreifzjut3te2nhyekklss27nh3k72ysco7y32koao5eei66wof36n5e",
+    "dataModelSelector": "Links/0/Links",
+    "activation": "2021-07-14T19:27:14.934572Z",
+    "created": "2021-07-14T19:27:14.934572Z",
+    "updated": "2021-07-14T19:27:14.934572Z"
```
[the remark this answers is not in the document]
Comment (alanshaw) [reply code-snippets/how-to/example-listing-upload.json:21]: https://github.com/web3-storage/web3.storage/issues/284

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/list.md`:
- The following link returned HTTP status code 302 (Found):
  https://web3.storage/files
- Possibly missing preposition found.
  ```
  ...nce you've stored some files using Web3.Storage, you'll want to see a list o...
                                             ^^^^^^^
  ```
- Possible typo. Did you mean “the” or “You”?
  ```
  ...ere are two ways you can do this: Using the  You can see a list of everythin...
                                             ^^^^^^^^
  ```
- Did you mean “these”?
  ```
  ...g the website may be a simpler choice.  This Files page  (opens new window) ...
                                             ^^^^
  ```
- Did you mean “them”?
  ```
  ...opens new window) that can be used with the for await syntax  (opens new win...
                                             ^^^
  ```
- Possible agreement error.
  ```
  ...an the total size of the files on disk. pins contains an array of objects de...
                                             ^^^^^^^^^^^^^
  ```
- Possible agreement error.
  ```
  ... fast retrieval using the IPFS network. deals contains an array of objects d...
                                             ^^^^^^^^^^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/list.md`:
- The following link returned HTTP status code 302 (Found):
  https://web3.storage/files
- Possibly missing preposition found.
  ```
  ...nce you've stored some files using Web3.Storage, you'll want to see a list o...
                                             ^^^^^^^
  ```
- Did you mean “these”?
  ```
  ...g the website may be a simpler choice.  This Files page  (opens new window) ...
                                             ^^^^
  ```
- Did you mean “them”?
  ```
  ...opens new window) that can be used with the for await syntax  (opens new win...
                                             ^^^
  ```
- Possible agreement error.
  ```
  ...an the total size of the files on disk. pins contains an array of objects de...
                                             ^^^^^^^^^^^^^
  ```
- Possible agreement error.
  ```
  ... fast retrieval using the IPFS network. deals contains an array of objects d...
                                             ^^^^^^^^^^^^^^
  ```

**Pull request merged**